### PR TITLE
build: fixes for cross-platform RBE

### PR DESCRIPTION
**Pull request by gregmagolan**
* patch for build_bazel_rules_nodejs from https://github.com/bazelbuild/rules_nodejs/pull/1320
* patch for @bazel/protractor from https://github.com/bazelbuild/rules_nodejs/pull/1320= (using patch-package npm tool for this as its much simpler to setup than the manual `postinstall-patches.js`)
* work-around for rules_webtesting cross-platform RBE issues

Cross-platform RBE on a mac with linux RBE workers requires the following config:

```
--config=remote --google_credentials=/path/to/credentials.json --cpu=k8 --host_cpu=k8
```

both `--cpu` & `--host_cpu` need to be specified when building from OSX or else there is bazel toolchain error:

```
ERROR: /private/var/tmp/_bazel_greg/5e8f8a9cd1c6fbc1afd11e37ee1fe2e5/external/bazel_toolchains/configs/ubuntu16_04_clang/9.0.0/bazel_1.0.0/cc/BUILD:50:1: in cc_toolchain_suite rule @bazel_toolchains//configs/ubuntu16_04_clang/9.0.0/bazel_1.0.0/cc:toolchain: cc_toolchain_suite '@bazel_toolchains//configs/ubuntu16_04_clang/9.0.0/bazel_1.0.0/cc:toolchain' does not contain a toolchain for cpu 'darwin'
```

Patches added here can be removed after https://github.com/bazelbuild/rules_nodejs/pull/1320 lands and angular is updated to a release of rules_nodejs that includes the changes.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Cross-platform RBE doesn't work.

Issue Number: https://github.com/bazelbuild/rules_nodejs/issues/1305

## What is the new behavior?
Cross-platform RBE works.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by gregmagolan** [comment]
@googlebot I consent

**Comment by gregmagolan** [comment]
Turned on Windows CI temporarily to verify it is green with this change: https://app.circleci.com/jobs/github/angular/angular/521453

**Review by petebacondarwin — APPROVED** [review]

**Comment by gregmagolan** [comment]
0.40 release is out which includes the changes in the patch here. After the https://github.com/angular/angular/pull/33802 update lands this PR can be updated and land without patches.

File: tools/defaults.bzl [hunk 1]
```diff
@@ -203,28 +204,22 @@ def karma_web_test_suite(bootstrap = [], deps = [], runtime_deps = [], **kwargs)
...
-
-    # rules_webtesting has a required_tag "native" for `chromium-local` browser
-    if not "native" in tags:
-        tags = tags + ["native"]
-
     _karma_web_test_suite(
         runtime_deps = local_runtime_deps,
         bootstrap = bootstrap,
         deps = local_deps,
         # Run unit tests on local Chromium by default.
```
Comment (josephperrott) [inline tools/defaults.bzl]: Can you remove this comment as firefox tests are no longe available with our move to use our own `browser_repositories.bzl`

**Review by josephperrott — APPROVED** [review]
LGTM, global approval

**Pull request closed without merging**